pandas/core/frame.py

```diff
@@ -1964,6 +1965,33 @@ def to_feather(self, path):
 
         to_feather(self, path)
 
+    def to_markdown(self, **kwargs):
+        """
+        Print a DataFrame in markdown-friendly format.
+
+        .. versionadded:: 1.0
+
+        Returns
+        -------
+        str
+            DataFrame in markdown-friendly format.
+
+        Examples
+        --------
+        >>> df = pd.DataFrame(data={'col1': [1, 2], 'col2': [3, 4]})
+        >>> print(df.to_markdown())
+        |    |   col1 |   col2 |
+        |---:|-------:|-------:|
+        |  0 |      1 |      3 |
+        |  1 |      2 |      4 |
+        """
+        tabulate = import_optional_dependency("tabulate")
+        if "headers" not in kwargs:
```

Comment:
This and the next line can be

```suggestion
        kwargs.setdefault("headers", "keys")
```

likewise for tablefmt.